system/Database/BaseBuilder.php

```diff
@@ -2385,7 +2385,7 @@ protected function _updateBatch(string $table, array $keys, array $values): stri
                 ",\n",
                 array_map(
                     static fn ($key, $value) => $key . ($value instanceof RawSql ?
-                        ' = ' . $value :
+                        ' = ' . str_replace('%', '%%', $value) :
```

Comment:
Why do we need to change the RawSql string?
I think if we change the string, it is not *raw*.

Reply:
```php
return sprintf($sql, $data);
```

Have to escape % which might show up in RawSql.

Reply:
If you use `sprintf()`, you must escape all `%` (except what you intended to replace), not only in RawSql strings.

Isn't it dangerous to use `sprintf()` to generate SQL strings?

Reply:
I wouldn't think if you did it properly. It is used quite a bit already. Most of the code where % is likely to show up is in the data to be placed. I just had to escape the RawSql.

Reply:
I sent a bug report: #6555

Reply:
I changed to str_replace. Its not often you use a % in table or column names but I suppose it could happen. I had added query caching to insertBatch as well so I went ahead and changed those as well.